src/components/BuySellForm/BuySellForm.tsx

```diff
@@ -103,11 +103,11 @@ const BuySellForm = ({ advertId, isModalOpen, onRequestClose }: TBuySellFormProp
     const counterpartyRateRef = useRef<number | undefined>(undefined);
 
     useEffect(() => {
-        if (isAdvertiser) {
+        if (isAdvertiser || advertiserPaymentMethods === undefined) {
             get();
         }
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [isAdvertiser]);
+    }, [isAdvertiser, advertiserPaymentMethods]);
```

Comment:
after going idle for some time, the p2p_advertiser_payment_methods data becomes undefined, its not due to BE overriding this. So i added this to recall advertiserPaymentMethods if it becomes undefined.